### PR TITLE
QOL-7906 Prepare for CKAN 2.9

### PR DESCRIPTION
**Pull request by ThrawnCA**
- Split Pylons vs Flask handling of response headers
- Allow POST requests to report view (to refresh the report)
- use more named routes for compatibility with both Pylons and Flask

### CODE REVIEW

File: ckanext/report/controllers/__init__.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ def report_view(report_name, organization=None, refresh=False):
         t.abort(401)
     except t.ObjectNotFound:
         t.abort(404)
+    except Exception as e:
+        log.error("Failed to get report: %s", e)
```
Comment (duttonw) [inline ckanext/report/controllers/__init__.py:42]: so what error should it be, or are you just letting it return a 500 or other error?

File: ckanext/report/controllers/__init__.py [hunk 2]
```diff
@@ -121,13 +124,14 @@ def report_view(report_name, organization=None, refresh=False):
...
                 t.abort(401)
             filename = 'report_%s.csv' % key
-            t.response.headers['Content-Type'] = 'application/csv'
-            t.response.headers['Content-Disposition'] = six.text_type('attachment; filename=%s' % (filename))
-            return make_csv_from_dicts(data['table'])
+            response_headers = {
+                'Content-Type': 'application/csv',
+                'Content-Disposition': six.text_type('attachment; filename=%s' % (filename))
+            }
+            return make_csv_from_dicts(data['table']), response_headers
```
Comment (duttonw) [inline ckanext/report/controllers/__init__.py:132]: 👍

File: ckanext/report/controllers/blueprints.py [hunk 3]
```diff
@@ -16,17 +18,28 @@ def redirect_to_index():
...
 reporting.add_url_rule(
     u'/reports', 'reports', view_func=redirect_to_index
 )
 reporting.add_url_rule(
-    u'/report/<report_name>', 'view', view_func=report_view
+    u'/report/<report_name>', 'view', view_func=view, methods=('GET', 'POST',)
 )
 reporting.add_url_rule(
-    u'/report/<report_name>/<organization>', 'org', view_func=report_view
+    u'/report/<report_name>/<organization>', 'org', view_func=view
```
Comment (duttonw) [inline ckanext/report/controllers/blueprints.py]: any methods?

File: ckanext/report/templates/report/view.html [hunk 4]
```diff
@@ -14,7 +14,7 @@ <h1>{{ report.title }}</h1>
           {{ _('Generated') }}: {{ h.report__render_datetime(report_date, '%d/%m/%Y %H:%M') }}
       </p>
       {% if c.userobj.sysadmin %}
-      <div class="panel panel-info" style="width=700px">
+      <div class="panel panel-info">
```
Comment (duttonw) [inline ckanext/report/templates/report/view.html:17]: maybe need another class so that style can be re-injected.

**Review by duttonw — CHANGES_REQUESTED** [review]

File: ckanext/report/controllers/__init__.py (hunk 1 above)
Comment (ThrawnCA) [reply ckanext/report/controllers/__init__.py:42]: It's just some extra logging.

File: ckanext/report/controllers/blueprints.py (hunk 3 above)
Comment (ThrawnCA) [reply ckanext/report/controllers/blueprints.py]: It defaults to GET, which should be all that's needed on that route. The report homepage needs POST for the refresh button.

Comment (ThrawnCA) [reply ckanext/report/controllers/blueprints.py]: My mistake, the org route needs POST as well.

File: ckanext/report/templates/report/view.html (hunk 4 above)
Comment (ThrawnCA) [reply ckanext/report/templates/report/view.html:17]: As far as I've seen, the style isn't needed on 2.8, and gets in the way on 2.9.

**Review by duttonw — APPROVED** [review]

**Pull request merged**